src/components/register/Register/index.jsx

```diff
@@ -0,0 +1,224 @@
+import { Alert, Button, Checkbox, Input } from "antd";
+import Modal from "antd/lib/modal/Modal";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { validateEmail, hasError } from "../registerHelpers";
+import ErrorMessage from "../ErrorMessage";
+
+const Register = ({
+  // check container/LoginRegister/index.jsx to see functions
+  onSubmit,
+  handleLogin,
+  handleFacebookAuth,
+  handleGoogleAuth,
+  // string
+  error,
+}) => {
+  const [t] = useTranslation();
+  const [registerInformation, setRegisterInformation] = useState({
+    fullName: "",
+    age: "",
+    email: "",
+    password: "",
+    isAgreed: false,
+  });
+  const [errors, setErrors] = useState({});
+  const [isTermsOpen, setIsTermsOpen] = useState(false);
+  const handleChange = (key, value) => {
+    const newValues = { ...registerInformation };
+    newValues[key] = value;
+    setRegisterInformation(newValues);
+  };
+
+  const newErrorsObj = (registerInfo) => {
+    const MAX_PASSWORD_LENGTH = 8;
+    const { fullName, email, age, password, isAgreed } = registerInfo;
+    const errorObj = {};
+    if (fullName.trim() === "") {
+      errorObj.fullName = t("register.fullName");
+    }
+    if (age.trim() === "") {
+      errorObj.age = t("register.age");
+    }
+    if (password.trim().length < MAX_PASSWORD_LENGTH) {
+      errorObj.password = t("register.passwordCharacter");
+    }
+    if (password.trim() === "") {
+      errorObj.password = t("register.enterPassword");
+    }
+    if (validateEmail(email).length === 0) {
+      errorObj.email = t("register.emailFormat");
+    }
+    if (!isAgreed) {
+      errorObj.isAgreed = t("register.agreeOnTerms");
+    }
+
+    return errorObj;
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (Object.keys(newErrorsObj).length > 0) {
```

Comment:
move inside of if to meaningful variable

Reply:
PTAL